blog-posts/boosting-developer-productivity-with-github-actions.md

```diff
@@ -0,0 +1,266 @@
+    const configFile = core.getInput('commitlintConfigFile');
+
+    core.info(`🔎 Checking if the title of this PR "${title}" meets the requirements ...`);
+
+    try {
+        const lintResult = await lint(title, configFile);
+        if (!lintResult.valid) {
+            core.setFailed(`\n ${formatResult(lintResult)}`);
+        } else {
+            core.info('✔️ All good');
+        }
+    } catch (error) {
+        core.setFailed(error as Error);
+    }
+})();
+```
+
+We won't go into a lot of detail about what this code does. I'll leave this as homework for you to explore.
+
+The main takeaway should be, that there are two important libraries that we are using: `@actions/core` and `@actions/github`. These two libraries provide a lot of utilities to interact with the process that is running on the GitHub runner. They allow you to fetch information about the event that triggered this workflow as well as make it easy to report the result of your action. You can get up close and personal with this code and explore the inner workings in [the action's repository](https://github.com/satellytes/commitlint-pr-title).
+
+### Compiling and bundling an Action
+As we've seen in the previous subsection we can have pretty much any code we want, as long as we provide a main entry point for the GitHub runner to execute. There are thousands of ways to turn TypeScript into JavaScript. The one way that I found most comfortable was to use [`@vercel/ncc`](https://github.com/vercel/ncc) to transpile, compile and bundle this action. `ncc` is a CLI tool that supports TypeScript out of the box and is meant to ship a single file that contains everything that is needed, from your actual source code to everything from inside the depths of your `node_modules` folder that is required by your action.
+
+The TypeScript file shown above is called `src/main.ts`. The bundling is done by calling `ncc build src/main.ts`. This will produce a file `dist/index.js` that we then point to in the `action.yml` file, as shown above.
+
+There are a few pitfalls here:
+- If your action involves a build step (like compiling TypeScript to JavaScript), you need to remember to include the generated bundle in your repository, in this case the `dist` folder. For regular projects this is not the case most of the time. Generally we don't want to include generated code in our repository. In this case you need to resist your urge to add `dist` to your `.gitignore` file.
+- You need to remember to run your build/bundle step whenever you change your source code and commit this artifact alongside your code changes. This is important for the compiled code in your repository to not get outdated. I've spent more time than I'm comfortable to admit, trying to figure out why a certain change didn't affect my action until I realized that I didn't recompile the code before committing.
+
+> 💡 Having a `pre-commit` hook that runs your build/bundle step is a great way to automate this.
```

Comment:
Very good point. The problem - even with plain JS - is, that it would still require a bundling-step to inline the code from the `node_modules` dependecies. Either way, I'll add another tip here, like you suggested.

Reply:
Good point with the bundling 👍 I really did not consider that. Thanks for the additional tip about that! This is resolved but I keep it open for my question:

What tool that is simple to use would you use in that case ? `esbuild` or `rollup` ? I mean even webpack is configurable to create a single bundle but the magic word here is configurable which we don't want to bother with here I guess?